src/raster/rasterband.rs

```diff
@@ -169,3 +183,85 @@ impl<'a> MajorObject for RasterBand<'a> {
 }
 
 impl<'a> Metadata for RasterBand<'a> {}
+
+/// # Read To Array
+/// A helper function that takes a raster band as input. It will
+/// read the rasterband into an ndarray.
+fn read_to_array<T: 'static + Copy + FromPrimitive>(rband: &RasterBand) -> Result<Array2<T>> {
+    // get the data type of the dataset
+    let gt: GDALDataType = rband.band_type();
+    // get the byte buffer of the full dataset
+    let buffer: Buffer<u8> = rband.read_band_as::<u8>()?;

```

Comment:
why do we need to read the data as `Buffer<u8>`?  `T: GDALDataType` allows reading the correct type.